addons/metrics-server/README.md

```diff
@@ -1,5 +1,7 @@
 # Kubernetes Metrics Server
 
+**This addon is deprecated. Set `spec.metricsServer.enabled: true` instead**
```

Comment:
Wouldn't it be good to also update the manifest's line anyway? https://github.com/kubernetes/kops/blob/174ba0e32338c828f1706e6845cf7da1c14d8707/addons/metrics-server/v1.16.x.yaml#L144 

Might need to add a new manifest though for a new version, not sure if this is getting backported

I would think that may make it easier for gradual transitions or just understanding that there's a difference between the two variants -- an important security one at that. Otherwise may make it seem like they're otherwise equivalent. I was using the `metrics-server` Helm Chart instead of the add-on as well, but looking at the add-on made me realize they both shared a security risk.

Reply:
This is a new feature, so it won't be backported. Adding a new manifest would also only be for 1.19+, so that kind of defeats the purpose of the deprecation notice.

But what I can do is add a note to the release notes. Should have done so anyway.

Reply:
Gotcha, makes sense. I was looking back into this due to some security settings (`kube-apiserver` not setting `--kubelet-certificate-authority`) and the release notes for #10836 , and it seems like I forgot to ever respond here. 

The [release](https://github.com/kubernetes/kops/releases/tag/v1.19.0) note on `metrics-server` doesn't say that the new add-on is more secure -- which is a really key piece. The wording makes it sound like the add-on is just done via the new approach now, so other than deprecation it doesn't motivate one to move to the new method as the security piece isn't explicitly mentioned. I was previously using `metrics-server` directly via its Helm chart, so I also wouldn't know that I could now remove `--kubelet-insecure-tls` from my config.
I think it would be useful to mention that it is more secure than the now deprecated variant and that users of direct `metrics-server` should remove the `--kubelet-insecure-tls` flag when upgrading to kOps 1.19+

Reply:
Good suggestions! Thanks! Would you be able to add a PR for this?

Reply:
It only took me 2 years to finally get the PR out (sorry -- but I did get several other PRs I had in my backlog out as well at least!) but finally made it: https://github.com/kubernetes/kops/pull/15327